Sources/PackageLoading/ToolsVersionParser.swift

```diff
@@ -85,7 +85,17 @@ public struct ToolsVersionParser {
             throw Error.nonUTF8EncodedManifest(path: manifestPath)
         }
 
-        return try self.parse(utf8String: manifestContentsDecodedWithUTF8)
+        do {
+            return try self.parse(utf8String: manifestContentsDecodedWithUTF8)
+        }
+        catch {
```

Comment:
catch specific error instead of nested if

Reply:
Does that work when we need to catch on specific arguments? I thought that was only possible at the type-level.

Reply:
this should work:

```swift
do {
  return try self.parse(utf8String: manifestContentsDecodedWithUTF8)
} catch Error.malformedToolsVersionSpecification(.commentMarker(.isMissing)) {
  throw UnsupportedToolsVersion(packageIdentity: .init(path: manifestPath), currentToolsVersion: .current, packageToolsVersion: .v3)
}
```